src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,135 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.exception.DeleteOutOfRangeException;
+import duke.exception.DoneOutOfRangeException;
+import duke.exception.EmptyDeadlineException;
+import duke.exception.EmptyEventException;
+import duke.exception.EmptyTodoException;
+import duke.exception.MissingDeadlineDateException;
+import duke.exception.MissingDeleteArgumentException;
+import duke.exception.MissingDoneArgumentException;
+import duke.exception.MissingEventDateException;
+import duke.exception.MissingFindArgumentException;
+import duke.exception.UnknownCommandException;
+
+/**
+ * Duke Bot is a program that c an schedule tasks and mark them as done, delete
+ * and save them on a file.
+ *
+ * @author Lim Jin Feng
+ * @version 0.1
+ * @since   2020-08-20
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs Duke Bot in a given file path.
+     *
+     * @param filePath  Where Duke Bot will save it's tasks.
+     */
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            this.tasks = new TaskList(storage.load());
+        } catch (FileNotFoundException e) {
+            this.ui.sendFailedInitialiseMessage();
+            this.tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Overloaded constructor that constructs Duke Bot with default file path,
+     * if it is not specified.
+     */
+    public Duke() {
+        String filePath = System.getProperty("user.dir") + "/data/Duke.txt";
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            this.tasks = new TaskList(storage.load());
+        } catch (FileNotFoundException e) {
+            this.ui.sendFailedInitialiseMessage();
+            this.tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs Duke Bot.
+     */
+    public void run() {
+        this.ui.sendInitialiseMessage();
+        this.ui.sendBar();
+        this.ui.greet();
+        this.ui.sendBar();
+        String userInput = this.ui.getUserInput();
+        while (!userInput.equals("bye")) {
```

Comment:
Perhaps you might want to use a boolean value 'isExit' as the looping condition instead? You may refer to the example below.

![image](https://user-images.githubusercontent.com/59356158/91817948-01d96600-ec68-11ea-8e11-d611405b14a6.png)